.github/workflows/ci.yml

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2023 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+name: bitcoin-core-ci
+run-name: Bitcoin Core CI
+# See: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore.
+on: [pull_request, push]
+
+env:
+  DANGER_RUN_CI_ON_HOST: 1
+  TEST_RUNNER_TIMEOUT_FACTOR: 40
+
+jobs:
+  macos-native:
+    name: macOS 13 native, x86_64 [no depends, sqlite only, gui]
+    # Use latest image, but hardcode version to avoid silent upgrades (and breaks).
+    # See: https://github.com/actions/runner-images#available-images.
+    runs-on: macos-13
+
+    env:
+      MAKEJOBS: '-j4'
+      CI_USE_APT_INSTALL: 'no'
+      PACKAGE_MANAGER_INSTALL: 'echo'  # Nothing to do
+      FILE_ENV: './ci/test/00_setup_env_mac_native_x86_64.sh'
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Clang version
+        run: clang --version
+
+      - name: Install Homebrew packages
+        run: brew install automake boost pkg-config miniupnpc libnatpmp zeromq qt@5 qrencode ccache gnu-getopt
+
+      - name: Ccache cache
+        uses: actions/cache@v3
+        with:
+          path: /Users/runner/work/bitcoin/bitcoin/ci/scratch/.ccache
+          key: ${{ runner.os }}-${{ runner.arch }}-ccache-cache-${{ github.run_id }}
+          restore-keys: ${{ runner.os }}-${{ runner.arch }}-ccache-cache
```

Comment:
> any reason to not just use the task name, like before?

GHA's job (an analogue of Cirrus's task) [context](https://docs.github.com/en/actions/learn-github-actions/contexts#job-context) has no "name"-like property.

Reply:
> Ok. You are setting one in the line above: `name: macOS 13 native, x86_64 [no depends, sqlite only, gui]`
> 
> But if that is not possible to use, that is fine.

Well, it is still possible to get `github.job` property that has value `macos-native` from https://github.com/bitcoin/bitcoin/blob/e9b72185205812df51082b42fb8c25cbf53cbc03/.github/workflows/ci.yml#L14-L15

> Though, it would be good to explain why both `key` and `restore-key` are set and why `github.run_id` is used?

This use case is described in `actions/cache` [docs](https://github.com/actions/cache/blob/main/tips-and-workarounds.md#update-a-cache). TL;DR: it forces to update the cache regardless of its successful restoring.

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/28187#issuecomment-1665619844).